gen3/tools/indexing/index_manifest.py

```diff
@@ -454,14 +492,54 @@ def _index_record(indexclient, replace_urls, thread_control, fi):
 
     except Exception as e:
         # Don't break for any reason
+        index_success = False
         exc_info = sys.exc_info()
         traceback.print_exception(*exc_info)
         logging.error(
-            "Can not update/create an indexd record with guid {}. Detail {}".format(
+            "Can not update/create an indexd record with guid {}. Detail: {}".format(
                 fi.get(GUID_STANDARD_KEY), e
             )
         )
 
+    # submit package metadata to metadata service
+    if index_success and fi.get(RECORD_TYPE_STANDARD_KEY) == "package":
+        try:
+            if not mds:
+                raise Exception(
+                    "Can not create package metadata when using indexd basic auth"
+                )
+
+            url = doc.urls[0]  # assume single URL for packages
+            file_name = doc.file_name if doc.file_name else os.path.basename(url)
```

Comment:
:shrug: idk which I actually like better. This one is debateable more readable.
```suggestion
            file_name = doc.file_name or os.path.basename(url)
```